src/main/webapp/app/overview/course-conversations/course-conversations.component.ts

```diff
@@ -46,15 +46,36 @@ export class CourseConversationsComponent implements OnInit, OnDestroy {
             .pipe(takeUntil(this.ngUnsubscribe))
             .subscribe({
                 complete: () => {
+                    this.subscribeToQueryParameter();
                     // service is fully set up, now we can subscribe to the respective observables
                     this.subscribeToActiveConversation();
                     this.subscribeToConversationsOfUser();
                     this.subscribeToLoading();
                     this.isServiceSetUp = true;
+                    this.updateQueryParameters();
                 },
             });
     }
 
+    subscribeToQueryParameter() {
+        this.activatedRoute.queryParams.pipe(take(1), takeUntil(this.ngUnsubscribe)).subscribe((queryParams) => {
+            if (queryParams.conversationId) {
+                this.metisConversationService.setActiveConversation(Number(queryParams.conversationId));
+            }
+        });
+    }
+
+    updateQueryParameters() {
+        this.router.navigate([], {
```

Comment:
Would it be possible to use `.../messages/6` instead of `.../messages?conversationId=6`?

Reply:
Its not a subpage it is to save the state

Reply:
@ge65cer Are you ok with keeping query parameters? We would like to have these fixed in 6.0

Reply:
I'd prefer using the first approach (but this is debatable and no deal-breaker), since we don't use query parameters anywhere else. @Strohgelaender seems to agree, any other opinions on this?

Reply:
I think the current solution by @stefanwaldhauser is ok, or is there a particular reason for your preference @ge65cer ?